feature-libs/product/configurators/rulebased/root/interceptor/cpq-access-storage.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Inject, Injectable, InjectionToken } from '@angular/core';
+import { Observable, timer } from 'rxjs';
+import { expand, filter, shareReplay, switchMap } from 'rxjs/operators';
+import { CpqAccessData } from './cpq-access-data.models';
+import { CpqAccessLoaderService } from './cpq-access-loader.service';
+
+const ONE_DAY = 24 * 60 * 60 * 1000;
+
+@Injectable({ providedIn: 'root' })
+export class CpqAccessStorageService {
```

Comment:
How can we trigger a new re-initialization of the token, e.g. if logout/login occured?
(we discussed this several times, one way would be to manage the token via a BehaviourSubject, there might be others)

Reply:
@ChristophHi Token fetching is  now started/stopped depending on login state.
Logging out will now unsubsribe from the token-access service, while logging in will subscribe to it.
subscribing to the login state is triggered by the first attempt to fetch a token, so this happens only if cpqConfigurator is used.